### ISSUE TITLE
Editor breaks when loading a project with too many scene files in a folder

### ISSUE THREAD

**Issue by AlbuquerqueSenior** [godotengine/godot#65536]
### Godot version

4.0 alpha16

### System information

Linux Ubuntu KDE, Nvidia RTX 2060

### Issue description

Editor breaks when loading a project with too many scene files in a folder.

### Steps to reproduce

I have a project with several chunks of scenes in a directory of my project. **10 thousand small files to be exact!** Each file on average 540kb. 

Until version 3.5 of Godot the project opened normally, but when I went to test on version 4.0 the editor broke.
I tested ignoring the directory contents with **.gdignore** file and the project opens normally without problems.

### Minimal reproduction project

Create a lot of scene files in a directory and Godot closes when it starts.

**Comment by KoBeWi** [comment]
Did you run the converter or are you opening an unconverted project? If the latter, the issue might be caused by too many errors.

What prints in the console when crash occurs?

**Comment by AlbuquerqueSenior** [comment]
I didn't convert a project, I just regenerated the scene parts with my own generation tool made in GDscript. It is a script that generates scenes procedurally. 

I did another test putting only a thousand files in the directory and the editor opens. It doesn't show anything on the console because the editor abruptly closes.

Another interesting detail is that the editor only closes when there are many ".scn" ou ".tscn" files. If you have ".txt" files the editor does not close!
